stl/inc/format

```diff
@@ -1496,6 +1496,218 @@ public:
     }
 };
 
+template <class _Ty>
+class _Fmt_buffer {
+private:
+    _Ty* _Ptr_        = nullptr;
+    size_t _Size_     = 0;
+    size_t _Capacity_ = 0;
+
+protected:
+    explicit _Fmt_buffer(const size_t _Size) noexcept : _Size_(_Size), _Capacity_(_Size) {}
+
+    ~_Fmt_buffer() = default;
+
+    _Fmt_buffer(_Ty* _Data, const size_t _Size, const size_t _Capacity) noexcept
+        : _Ptr_(_Data), _Size_(_Size), _Capacity_(_Capacity) {}
+
+    void _Set(_Ty* _Buf_data, const size_t _Buf_capacity) noexcept {
+        _Ptr_      = _Buf_data;
+        _Capacity_ = _Buf_capacity;
+    }
+
+    virtual void _Grow(size_t _Capacity) = 0;
+
+public:
+    using value_type = _Ty;
+
+    _Fmt_buffer(const _Fmt_buffer&) = delete;
+    void operator=(const _Fmt_buffer&) = delete;
+
+    _NODISCARD _Ty* begin() noexcept {
+        return _Ptr_;
+    }
+
+    _NODISCARD _Ty* end() noexcept {
+        return _Ptr_ + _Size_;
+    }
+
+    _NODISCARD size_t _Size() const noexcept {
+        return _Size_;
+    }
+
+    _NODISCARD size_t _Capacity() const noexcept {
+        return _Capacity_;
+    }
+
+    void _Clear() noexcept {
+        _Size_ = 0;
+    }
+
+    void _Try_resize(const size_t _Count) {
+        _Try_reserve(_Count);
+        _Size_ = _Count <= _Capacity_ ? _Count : _Capacity_;
+    }
+
+    void _Try_reserve(const size_t _New_capacity) {
+        if (_New_capacity > _Capacity_) {
+            _Grow(_New_capacity);
+        }
+    }
+
+    void push_back(const _Ty _Value) {
+        _Try_reserve(_Size_ + 1);
+        _Ptr_[_Size_++] = _Value;
+    }
+};
+
+struct _Fmt_buffer_traits {
+    explicit _Fmt_buffer_traits(ptrdiff_t) {}
+
+    _NODISCARD size_t _Count() const noexcept {
+        return 0;
+    }
+
+    _NODISCARD size_t _Limit(const size_t _Size) noexcept {
+        return _Size;
+    }
+};
+
+class _Fmt_fixed_buffer_traits {
+private:
+    ptrdiff_t _Count_ = 0;
+    ptrdiff_t _Limit_;
+
+public:
+    explicit _Fmt_fixed_buffer_traits(const ptrdiff_t _Limit) noexcept : _Limit_(_Limit) {}
+
+    _NODISCARD size_t _Count() const noexcept {
+        return static_cast<size_t>(_Count_);
+    }
+
+    _NODISCARD size_t _Limit(const size_t _Size) noexcept {
+        size_t _Avail = static_cast<size_t>(_Limit_ > _Count_ ? _Limit_ - _Count_ : 0);
+        _Count_ += _Size;
+        return _Size < _Avail ? _Size : _Avail;
+    }
+};
+
+inline constexpr size_t _Fmt_buffer_size = 256;
+
+template <class _OutputIt, class _Ty, class _Traits = _Fmt_buffer_traits>
+class _Fmt_iterator_buffer final : public _Traits, public _Fmt_buffer<_Ty> {
+private:
+    _OutputIt _Output;
+    _Ty _Data[_Fmt_buffer_size];
+
+    void _Grow(size_t) final {
+        if (this->_Size() == _Fmt_buffer_size) {
+            _Flush();
+        }
+    }
+
+    void _Flush() {
+        auto _Size = this->_Size();
+        this->_Clear();

```

Comment:
This is correct, but super subtle and requires the user to check the implementation of `_Clear` to verify that all is fine.

Can we move the `this->_Clear()` call below the `_Copy_unchecked` call so that it is obvious that we do not delete what we want to copy